components/base/kinematicbase/ptgKinematics.go

```diff
@@ -133,11 +133,22 @@ func wrapWithPTGKinematics(
 		origin = originPIF.Pose()
 	}
 
+	// construct executionFrame
+	executionFrame, err := referenceframe.New2DMobileModelFrame(
```

Comment:
This should be a full 6dof pose, not a 2d pose.

Reply:
I think its premature to make this change there is no use case yet where we need more DOF than what exists for (x, y, theta)

Reply:
Creating such a frame would require construction of a new frame with inputs [x, y, z, roll, pitch, yaw] and since we would only be using 3 of those for now it seems unnecessary to introduce this code at this moment

Reply:
I disagree.

Consider a 2d robot rolling over a bump with a lidar. It may be temporarily looking up or down, and seeing bonus obstacles that are actually the floor or ceiling. If we capture the pose of the robot, we are able to account for this.

As written we are throwing information away and actively weakening our current spatial representation. Right now this can be captured.